staging-service/src/main/java/com/github/cafdataprocessing/services/staging/utils/JsonMinifier.java

```diff
@@ -94,14 +101,17 @@ private static void processJsonTokens(final JsonParser parser,
                                           final JsonGenerator gen,
                                           final String storageRefPath,
                                           final String inprogressContentFolderPath,
-                                          final int fieldValueSizeThreshold) throws IOException {
+                                          final int fieldValueSizeThreshold,
+                                          final Set<String> binaryFilesUploaded) throws IOException, InvalidBatchException {
         String dataBuffer = null;
         String encodingBuffer = null;
         JsonToken token;
         boolean pauseWriting = false;
         boolean bufferData = false;
         boolean bufferEncoding = false;
         boolean updateReference = false;
+        // mark if the updateReference is for a file that has been uploaded
+        boolean isLocalRefFile = false;
```

Comment:
Is this definitely needed?  How is it different from `updateReference`?

Reply:
`updateReference` is used to mark that a `local_ref` has been modified as `storage_ref`, but also in the case where the content of the [dataBuffer exceeds a threshold and is written to a file](https://github.com/CAFDataProcessing/staging-service/blob/afe3bf570e255ba79d4c454565608888cf4ce573/staging-service/src/main/java/com/github/cafdataprocessing/services/staging/utils/JsonMinifier.java#L159). 

In this latter case, if the code checks to see if `updateReference` is true [here](https://github.com/CAFDataProcessing/staging-service/blob/afe3bf570e255ba79d4c454565608888cf4ce573/staging-service/src/main/java/com/github/cafdataprocessing/services/staging/utils/JsonMinifier.java#L168), and then checks if the file has been uploaded [here](https://github.com/CAFDataProcessing/staging-service/blob/afe3bf570e255ba79d4c454565608888cf4ce573/staging-service/src/main/java/com/github/cafdataprocessing/services/staging/utils/JsonMinifier.java#L169), it will return false, because the file that has been created to write the buffer is not on the list of files uploaded.

Therefore, I just added another flag with a different meaning for this specific case.